test/metabase/query_processor_test/date_bucketing_test.clj

```diff
@@ -1163,6 +1163,19 @@
                         :breakout    [!day-of-week.date]
                         :filter      [:between $date "2013-01-03" "2013-01-20"]}))))))))))
 
+(deftest first-day-of-week-for-day-of-week-bucketing-test
+  (testing "First day of week for `:day-of-week` bucketing should be the consistent (#17801)"
+    (mt/test-drivers (mt/normal-drivers-with-feature :basic-aggregations)
+      (let [query (mt/mbql-query checkins
+                    {:aggregation [[:count]]
+                     :breakout    [!day-of-week.date]})]
+        (doseq [[first-day-of-week expected-rows] {:sunday [[1 135] [2 143] [3 153] [4 136] [5 139] [6 160] [7 134]]
```

Comment:
is it worth doing this for all days of the week?
```clojure
;; either generate it in code
(zipmap [:sunday :monday :tuesday :wednesday :thursday :friday :saturday]
        (map (fn [n coll]
                (mapv vector (next (range 7)) (drop n coll)))
              (range 7)
              (repeat (cycle [135 143 153 136 139 160 134]))))

;; or use the results of it cycling through.
{:sunday    [[1 135] [2 143] [3 153] [4 136] [5 139] [6 160]],
 :monday    [[1 143] [2 153] [3 136] [4 139] [5 160] [6 134]],
 :tuesday   [[1 153] [2 136] [3 139] [4 160] [5 134] [6 135]],
 :wednesday [[1 136] [2 139] [3 160] [4 134] [5 135] [6 143]],
 :thursday  [[1 139] [2 160] [3 134] [4 135] [5 143] [6 153]],
 :friday    [[1 160] [2 134] [3 135] [4 143] [5 153] [6 136]],
 :saturday  [[1 134] [2 135] [3 143] [4 153] [5 136] [6 139]]}
```

Reply:
Hmmm. Not a bad idea. I'll have to update that

Reply:
fixed